### PR TITLE
chore(j-s): Prosecutors Office

### PR DESCRIPTION
**Pull request by gudjong**
# Prosecutors Office

[Geyma embætti stofnanda sérstaklega í RVG](https://app.asana.com/0/1199153462262248/1206153194238639/f)

## What

- Stores the prosecutors office id directly in the case.

## Why

- To make sure the case stays with the prosecutoras office even when the creating prosecutor moves to a different office.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts [hunk 1]
```diff
@@ -463,6 +463,7 @@ export class InternalCaseService {
             prosecutorId:
               creator.role === UserRole.PROSECUTOR ? creator.id : undefined,
             courtId: creator.institution?.defaultCourtId,
+            prosecutorsOfficeId: creator.institution?.id,
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts:466]: Please remove Institution and User imports as they are no longer being used.

File: apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts [hunk 2]
```diff
@@ -587,6 +575,15 @@ export class InternalCaseService {
             this.config.archiveEncryptionKey,
             { iv: CryptoJS.enc.Hex.parse(uuidFactory()) },
           ).toString(),
+          // To decrypt:
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts:578]: Is this meant to be a part of this PR?

File: apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/create.spec.ts [hunk 3]
```diff
@@ -97,11 +97,13 @@ describe('InternalCaseController - Create', () => {
 
   describe('creating case', () => {
     const userId = uuid()
+    const prosecutorsOfficeId = uuid()
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/case/test/internalCaseController/create.spec.ts:100]: Please remove Institution import as it's no longer being used.

**Review by oddsson — APPROVED** [review]
Nice! There are just a couple of unused imports..

File: apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts:578]: Yes, every time we need to decrypt the arvhived information we have had a hard time figuring out how.

**Pull request merged**